org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/grammarAccess/GrammarAccessExtensions.xtend

```diff
@@ -396,8 +392,49 @@ class GrammarAccessExtensions {
 		}
 	}
 	
-	def String grammarFragmentToString(EObject ele, String prefix) {
-		return serializer.grammarFragmentToString(ele, prefix)
+	def String grammarFragmentToString(EObject object, String prefix) {
+		val node = NodeModelUtils.findActualNodeFor(object)
+		if (node === null) {
+			if (object instanceof RuleCall)
+				if (object?.rule?.name !== null)
+					return process(object.rule.name, prefix)
+			return ""
+		} else {
+			return node.text.process(prefix)
+		}
+	}
+
+	private def process(String input, String prefix) {
+		// remove leading and trailing blank lines
+		var lines = input.split('(\\r?\\n)')
+		var first = 0
+		while (isBlank(lines.get(first)))
+			first++
+		var last = lines.length - 1
+		while (isBlank(lines.get(last)))
+			last--
+		lines = lines.subList(first, last + 1)
+		// remove common whitespace (e.g. leading blanks) and add prefix to each line
+		val commonWhitespace = commonLeadingWhitespace(lines)
+		for (var n=0; n<lines.length; n++)
+			lines.set(n, prefix + lines.get(n).substring(commonWhitespace.length))
+		// generate result by joining lines
+		return lines.join(System.lineSeparator)
```

Comment:
This should use the line ending from the code config

Reply:
Ok.